### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by PeterDaveHello**
Add info about shallow clone, to save the bandwidth, disk space, and time!

### CODE REVIEW

**Comment by ihnorton** [comment]
-1. We want people to be able to contribute and the difference is
negligible given dependencies.

**Comment by PeterDaveHello** [comment]
So they have their choice, I didn't jut replace the old one, but add a new method. 
If they are not going to contributing, full cloning is just wasting the resource.

**Comment by tkelman** [comment]
Note that we have a formatting convention for both code and documentation to avoid trailing whitespace, which is why this failed on Travis CI.

Don't shallow clones have a bunch of limitations?

**Comment by PeterDaveHello** [comment]
@tkelman, thanks, I'll check it, I used GitHub's editor to edit the online, will not clone it to fix.
Shallow clone has a limitation, is that you won't have the history of that project, but for non-developer, that's no hurt, it won't bite anyone.

**Comment by tkelman** [comment]
If you're cloning the source, you probably want to participate in development? Can you still do full rebases, merges, cherry-picks, what have you? If any common task that comes up while, say, working on a pull request can't be done with a shallow clone, I don't think we should recommend it.

File: README.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ First, acquire the source code by cloning the git repository:
 
     git clone git://github.com/JuliaLang/julia.git
 
+If you are not going to joing the development but only need the source for compilation, which means you don't need the commit history, add parameter `--depth 1` to shallow clone, it will really help you speed up the cloning process, and save the bandwidth and disk space:
```
Comment (tkelman) [inline README.md]: some types - "joing" should be "join," and no need for the article "the" in "save bandwidth and disk space."

**Comment by PeterDaveHello** [comment]
Actually, I did this PR by shallow clone, it's fast, and I don't need the history, that part ** Source Download and Compilation** is not talking about development, and I think the developers know what they should do. Any way, this is my point, I respect your decision.

File: README.md (hunk 1 above)
Comment (PeterDaveHello) [inline README.md]: Thanks.

**Comment by PeterDaveHello** [comment]
Fixed with shallow clone haha ...

**Comment by tkelman** [comment]
> Fixed with shallow clone haha ...

In that case I change my mind, I guess this doesn't hurt to mention as an option. If you add a note that you can convert from a shallow clone to a full one by running `git fetch --unshallow`, then I think this would be fine.

**Comment by ScottPJones** [comment]
I think this is useful information, even for contributors... for example, I've got clones of Julia on a desktop, on my laptop, and on some VMs for Linux (and I'll need to make another for Windows)...  I really only want all the history on the one that I actually use to create branches & PRs, this could save me some space...

**Comment by tkelman** [comment]
Fair enough, I'm convinced:
```
tkelman@ygdesk:~/Julia$ git clone https://github.com/JuliaLang/julia --depth=1 julia-shallow
Cloning into 'julia-shallow'...
remote: Counting objects: 922, done.
remote: Compressing objects: 100% (860/860), done.
remote: Total 922 (delta 16), reused 232 (delta 11), pack-reused 0
Receiving objects: 100% (922/922), 2.89 MiB | 1.72 MiB/s, done.
Resolving deltas: 100% (16/16), done.
Checking connectivity... done.
tkelman@ygdesk:~/Julia$ git clone https://github.com/JuliaLang/julia julia-deep
Cloning into 'julia-deep'...
remote: Counting objects: 133351, done.
remote: Compressing objects: 100% (107/107), done.
remote: Total 133351 (delta 42), reused 0 (delta 0), pack-reused 133244
Receiving objects: 100% (133351/133351), 69.62 MiB | 9.69 MiB/s, done.
Resolving deltas: 100% (100813/100813), done.
Checking connectivity... done.
tkelman@ygdesk:~/Julia$ du -h --max-depth=0 julia-shallow
16M     julia-shallow
tkelman@ygdesk:~/Julia$ du -h --max-depth=0 julia-deep
87M     julia-deep
```

**Comment by PeterDaveHello** [comment]
Done.

**Comment by ViralBShah** [comment]
For people who are bandwidth constrained, why not point them to the tarball download instead?

**Comment by PeterDaveHello** [comment]
> For people who are bandwidth constrained, why not point them to the tarball download instead?
Because tarball doesn't need instruction? 
Shallow clone can still work, only without the history, still can doing development things, and commit, push, but tarball doesn't.

**Comment by ihnorton** [comment]
Note that `--unshallow` is not available until git 1.8.3.

**Comment by garrison** [comment]
Also, I believe @kshyatt just discovered that cloning with `--depth 1` on master will result in a `julia` binary whose `VERSION` is `0.4.0-dev` without the revision number (e.g. instead of `0.4.0-dev+4462`).  This then breaks `Compat.jl` and other version conditionals.  For this reason alone we should probably not recommend that people make shallow clones of the repository.

**Comment by tkelman** [comment]
sold, closing then. ref #10851

**Pull request closed without merging**